packages/common/src/hooks/usePagination/usePagination.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
+
+interface PaginationState {
+  page: number;
+  offset: number;
+  first: number;
+  onChangePage: (newPage: number) => void;
+  onChangeFirst: (newFirst: number) => void;
+}
+
+export const usePagination = (initialFirst = 20): PaginationState => {
+  const [first, onChangeFirst] = useState(initialFirst);
```

Comment:
Possibly need some defensive code to ensure `first` is > 0

Reply:
I can also change the `page` and `first` to negative numbers, I guess ideally we'd prevent some of that.

Reply:
Ah yes! Do you think `if < valid number` throw an error or default to the minimum?

Reply:
I was thinking to default to a minimum - but now that you say it, I rather like the idea of throwing an error. Otherwise we'll just carry a silent bug around when someone accidentally goes negative with one of those values.